### PR TITLE
tests/MM-T1128 : Announcement Banner - Dismissible banner shows long text truncated

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Ticket Link

https://automation-test-cases.vercel.app/test/MM-T1128

#### Related Pull Requests

- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import {hexToRgbArray, rgbArrayToString} from '../../../utils';
+
+describe('Announcement Banner', () => {
+    it('MM-T1128 Announcement Banner - Dismissible banner shows long text truncated', () => {
+        const bannerText =
+            "Here's an announcement! It has a link: http://example.com. It's a really long announcement, because we have a lot to say. Be sure to read it all, click the link, then dismiss the banner, and then you can go on to the next test, which will have a shorter announcement. Thank you for reading and have a nice day!";
```
Comment (M-ZubairAhmed) [inline e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js:15]: > todo change this to bannerEmbeddedLink

File: e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            and(
+                'have.css',
+                'background-color',
+                rgbArrayToString(bannerBgColorRGBArray),
+            ).
+            and(
+                'have.css',
+                'color',
+                rgbArrayToString(bannerTextColorRGBArray),
+            );
```
Comment (M-ZubairAhmed) [inline e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js:74]: i missed one point from spec
> URL is a live link
i think if i just prove that anchor tag with href exists ?

File: e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        cy.visit('/admin_console/site_config/announcement_banner');
+
+        // # Enable banner if not already enabled
+        cy.findByTestId('AnnouncementSettings.EnableBanner').within(() => {
+            cy.findByText('true').should('be.visible').click({force: true});
+        });
+
+        // # Enter the long banner text
+        cy.findByTestId('AnnouncementSettings.BannerText').within(() => {
+            cy.get('input').should('be.visible').clear().type(bannerText);
```
Comment (M-ZubairAhmed) [inline e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js:34]: when i wrote the invoke type for entering the text, the input text was vanishing as soon as input lost focus, 
so i had to resort to type methods

**Review by hmhealey — APPROVED** [review]
Looks good to me assuming you've addressed the things you wanted to in those comments.

File: e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js (hunk 3 above)
Comment (M-ZubairAhmed) [reply e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js]: figured it out, had to add type at the end

File: e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js (hunk 2 above)
Comment (M-ZubairAhmed) [reply e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js]: added the url test too

File: e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js (hunk 1 above)
Comment (M-ZubairAhmed) [reply e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js]: done

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @M-ZubairAhmed! Tested and passed. LGTM, except for one comment.

Note: I filed https://mattermost.atlassian.net/browse/MM-36095 to address UI polish in showing Announcement banner.

File: e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import {hexToRgbArray, rgbArrayToString} from '../../../utils';
+import * as TIMEOUTS from '../../../fixtures/timeouts';
+
```
Comment (saturninoabril) [inline e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js:12]:
```suggestion
import * as TIMEOUTS from '../../../fixtures/timeouts';

// Group: @enterprise @system_console @announcement_banner

```

**Review by josephbaylon — CHANGES_REQUESTED** [review]
Overall looks good. Please apply Saturn's latest suggestions. Thanks!

File: e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+describe('Announcement Banner', () => {
```
[the remark this answers is not in the document]
Comment (M-ZubairAhmed) [reply e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js:15]: ✅  Updated as per request/discussion

File: e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    it('MM-T1128 Announcement Banner - Dismissible banner shows long text truncated', () => {
+        const bannerEmbedLink = 'http://example.com';
+        const bannerText = `Here's an announcement! It has a link: ${bannerEmbedLink}. It's a really long announcement, because we have a lot to say. Be sure to read it all, click the link, then dismiss the banner, and then you can go on to the next test, which will have a shorter announcement. Thank you for reading and have a nice day!`;
+        const bannerBgColor = '#4378da';
+        const bannerBgColorRGBArray = hexToRgbArray(bannerBgColor);
+        const bannerTextColor = '#ffffff';
+        const bannerTextColorRGBArray = hexToRgbArray(bannerTextColor);
+
+        // # Login as System Admin
+        cy.apiAdminLogin();
```
[the remark this answers is not in the document]
Comment (M-ZubairAhmed) [reply e2e/cypress/integration/system_console/site_configuration/announcement_banner_spec.js]: ✅  Updated as per request/discussion

**Review by josephbaylon — APPROVED** [review]
LGTM. Thanks @M-ZubairAhmed !!

**Comment by josephbaylon** [comment]
@saturninoabril I know you're out in the next 2 weeks but this PR won't allow me to merge unless you approve of it. Please take a look if you have the chance. Thanks!

**Review by saturninoabril — APPROVED** [review]
Thanks @M-ZubairAhmed!

**Pull request merged**